src/scripts/create_icelandic_error_corpus.py

```diff
@@ -0,0 +1,133 @@
+"""Create the Icelandic Error Corpus dataset (a subset and the full dataset) and upload it to the HF Hub."""
+
+import re
+from typing import List
+
+from datasets import Dataset, DatasetDict, Split, load_dataset
+from huggingface_hub import HfApi
+from requests.exceptions import HTTPError
+
+
+def main():
+    """Create the Icelandic Error Corpus dataset (a subset and the full dataset) and upload it to the HF Hub."""
+    repo_id = "mideind/icelandic-error-corpus-IceEC"
+    dataset = load_dataset(path=repo_id, name="category")
+
+    train_df = prepare_dataframe(dataset=dataset["train"])
+    test_df = prepare_dataframe(dataset=dataset["test"])
+
+    # Make validation split
+    val_size = 1024
+    val_df = train_df.sample(n=val_size, random_state=4242)
+    train_df = train_df.drop(val_df.index)
+
+    dataset = DatasetDict(
+        train=Dataset.from_pandas(train_df, split=Split.TRAIN, preserve_index=False),
+        val=Dataset.from_pandas(val_df, split=Split.VALIDATION, preserve_index=False),
+        test=Dataset.from_pandas(test_df, split=Split.TEST),
+    )
+
+    # Make subset of the dataset with 1024 train samples, 256 validation samples, and 2048 test samples
+    dataset_subset = DatasetDict(
```

Comment:
Just adding a comment justifying the use of `head`.

```suggestion
    # We use `head` instead of `sample` here as the dataframes have already been shuffled.
    dataset_subset = DatasetDict(
```